doc/source/groups_of_atoms.rst

```diff
@@ -0,0 +1,117 @@
+
+These are considered atom :ref:`topology attributes <topology-attributes>`.
+
+The corresponding container groups are :class:`~MDAnalysis.core.groups.ResidueGroup` and :class:`~MDAnalysis.core.groups.SegmentGroup`. These have similar properties and available methods as :class:`~MDAnalysis.core.groups.AtomGroup`.
+
+Each of these container groups can be accessed through another. For example:
+
+.. code-block::
+
+    >>> ag = u.atoms.select_atoms('resname ARG and name CA')
+    >>> ag
+    <AtomGroup with 13 atoms>
+    >>> ag.residues
+    <ResidueGroup with 13 residues>
+    >>> ag.residues.atoms
+    <AtomGroup with 312 atoms>
+    >>> ag.segments
+    <SegmentGroup with 1 segment>
+    >>> ag.segments.atoms
+    <AtomGroup with 3341 atoms>
+
+Similarly, an :class:`~MDAnalysis.core.groups.Atom` has direct knowledge of the :class:`~MDAnalysis.core.groups.Residue` and :class:`~MDAnalysis.core.groups.Segment` it belongs to.
+
+.. code-block::
+
+    >>> a = u.atoms[0]
+    >>> a.residue
+    <Residue MET, 1>
+    >>> a.residue.segment
+    <Segment 4AKE>
+    >>> a.residue.segment.residues
+    <ResidueGroup with 214 residues>
+
+Adding to a topology
+--------------------
+
+To add a :code:`Residue` or :code:`Segment` to a topology, use the :meth:`Universe.add_Residue <MDAnalysis.core.universe.Universe.add_Residue>` or :meth:`Universe.add_Segment <MDAnalysis.core.universe.Universe.add_Segment>` methods.
+
+.. code-block::
+
+    >>> u = mda.Universe(PSF, DCD)
+    >>> u.segments
+    <SegmentGroup with 1 segment>
+    >>> u.segments.segids
+    array(['4AKE'], dtype=object)
+    >>> newseg = u.add_Segment(segid='X')
+    >>> u.segments.segids
+    array(['4AKE', 'X'], dtype=object)
+    >>> newseg.atoms
+    <AtomGroup with 0 atoms>
+
+To assign the last 100 residues from the :class:`~MDAnalysis.core.universe.Universe` to this new Segment:
+
+.. code-block::
+
+    >>> u.residues[-100:].segments = newseg
+    >>> newseg.atoms
+    <AtomGroup with 1600 atoms>
+
+Another example is `creating custom segments for protein domains <examples/constructing_universe.ipynb#Adding-a-new-segment>`_.
+
+---------------------------
+Groups of atoms in analysis
+---------------------------
+
+Certain analysis methods in MDAnalysis also make use of additional ways to group atoms.
+
+Fragments
+---------
+
+A fragment is defined by bond connectivity. A fragment is what is typically considered a molecule: a group of atoms where each atom is bonded to at least one other atom in the fragment, and are not bonded to any atoms outside the fragment. The fragments of a Universe are determined by MDAnalysis as a derived quantity.
+
+The fragments of an :class:`~MDAnalysis.core.groups.AtomGroup` are accessible via the :attr:`fragments` property. In the case below, there is only one fragment in the Universe:
+
+.. code-block::
+
+    >>> u.atoms.fragments
+    (<AtomGroup with 3341 atoms>,)
+
+Molecules
```

Comment:
"Molecules" is another "advanced" section that distracts from the important things. I would put it in an advanced topolgy chapter and footnote/link.

Reply:
Fair enough. "What is a molecule?" was one of the immediate questions I had when I was looking at AtomGroup's split, accumulate, center/centroid methods (and "why isn't it in the other methods?") but hyperlinks and the search bar hopefully make this a non-issue.